### DISCUSSION TITLE
INVALID_URI error

### DISCUSSION THREAD

**Discussion by MichalKotecki** [Kurento/kurento#25] [Community]
I recently started getting an INVALID_URI error. This not only happens in my application, but also for the Java Player endpoint tutorial for various video URIs, so I don't think it is related to code, maybe this is something about  Kurento docker container I use, although I am not sure. 

o.kurento.tutorial.player.PlayerHandler  : ErrorEvent: Error code 4: Stream doesn't contain enough data., source: kmsplayerendpoint3, element: kmsplayerendpoint3, debug info: gsttypefindelement.c(983): gst_type_find_element_chain_do_typefinding (): /GstPipeline:internalpipeline/GstURIDecodeBin:uridecodebin3/GstTypeFindElement:typefindelement3:
Can't typefind stream
o.kurento.tutorial.player.PlayerHandler  : ErrorEvent: Invalid URI
o.kurento.tutorial.player.PlayerHandler  : ErrorEvent: Invalid URI

What does this error mean and how can I proceed to debug it.

**Comment by j1elo** [comment]
> What does this error mean and how can I proceed to debug it.

Please provide an example or couple examples of the URIs you are giving the PlayerEndpoint to play

**Reply by JalODan** [reply]
Hello Juan,
I am not sure I can provide a couple of examples, but one example is all I've got. 
I have created a test project, put a video under resources/static/videos.
![image](https://github.com/Kurento/kurento/assets/58619914/84fa4264-5f35-4b8e-b88b-282eaae11a56)
The uploaded video can be accessed via URL http://localhost:8080/videos/bunny.webm.
When I provided kurento with this URL, it failed to play it.
Logs are: 
`2023-11-15 12:35:43.321  INFO 7992 --- [EventExec-e2-t3] o.kurento.tutorial.player.PlayerHandler  : ErrorEvent: Invalid URI
2023-11-15 12:35:43.321  INFO 7992 --- [EventExec-e2-t2] o.kurento.tutorial.player.PlayerHandler  : ErrorEvent: Invalid URI
2023-11-15 12:35:43.321  INFO 7992 --- [EventExec-e2-t1] o.kurento.tutorial.player.PlayerHandler  : ErrorEvent: Error code 4: Stream doesn't contain enough data., source: kmsplayerendpoint7, element: kmsplayerendpoint7, debug info: gsttypefindelement.c(986): gst_type_find_element_chain_do_typefinding (): /GstPipeline:internalpipeline/GstURIDecodeBin:uridecodebin7/GstTypeFindElement:typefindelement7:
Can't typefind stream`.
Any help is appreciated.

**Comment by JalODan** [comment]
Hello Michael,
I have also encountered this problem. Any luck in solving it? If so, please share
